operatorcourier/validate.py

```diff
@@ -85,30 +93,27 @@ def _bundle_validation(self, bundle):
 
     def _crd_validation(self, bundleData):
         logger.info("Validating custom resource definitions.")
-        valid = True
-
         crds = bundleData[self.crdKey]
 
+        valid = True
         for crd in crds:
-            if "metadata" in crd:
-                if "name" in crd["metadata"]:
-                    logger.info("Evaluating crd %s", crd["metadata"]["name"])
-                else:
-                    self._log_error("crd metadata.name not defined.")
-                    valid = False
-            else:
-                self._log_error("crd metadata not defined.")
-                valid = False
-
-            if "apiVersion" not in crd:
-                self._log_error("crd apiVersion not defined.")
-                valid = False
+            for crd_required_nested_field in self.crd_required_nested_fields:
+                valid &= self._crd_fields_validation(crd_required_nested_field.split('.'), crd, "")
+        return valid
 
-            if "spec" not in crd:
-                self._log_error("crd spec not defined.")
-                valid = False
+    # TODO: refactor csv and pkg fields presence validations with this helper
```

Comment:
We shouldn't leave TODOs in our code unless it is something that is important for the documentation of the source code itself (for example, if what was committed was a necessary workaround that requires understanding later). In this case, this TODO just seems like another work item we should do later. Please omit it.